src/main/logutils.ts

```diff
@@ -262,6 +286,69 @@ const determineArenaMatchResult = (line: string): any[] => {
     recorder.stop(metadata, overrun);
 }
 
+/**
+ * Handle a line from the WoW log. 
+ */
+ const handleChallengeModeStartLine = (line: string) => {
+    /**
+     * starting a new keystone fires CHALLENGE_MODE_STOP before the CHALLENGE_MODE_START
+     * so in theory if you start a new keystone before a failed key has been stopped the
+     * video will still be stopped/started in quick succession to ensure a separation.
+     */
+
+    const encounterID = parseInt(line.split(',')[1]);
+    const category = "ChallengeModes";
+
+    videoStartDate = getCombatLogDate(line);
+
+    metadata = {
```

Comment:
we probably want to capture the key level too

this is actually why i stopped working on this. I originally was just going to add all this detail to this metadata but then i realized we are going to have alot of dead properties depending on the gameplay type. so the best way was to create an abstract class with the common attributes and then have other gameplay types extend that depending on their use case.

Reply:
That makes sense, the key level is certainly something useful to display to the user.